### PR TITLE
[GEOT-7340] Avoid PAMParser init causing ImageMosaic to fail

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7340](https://osgeo-org.atlassian.net/browse/GEOT-7340) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Fixes an annoying exception in static part of class initialization when using ImageMosaic.

Related to class loading of javax.xml.bind.JAXBException, and how this is treated in it.geosolutions.imageio.pam.PAMParser.

Not sure how to write a test case for this one.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java [hunk 1]
```diff
@@ -153,10 +153,9 @@ public URL resolve(String granuleLocation) {
...
                     new VectorBinarizeRIF(),
                     Registry.JAI_TOOLS_PRODUCT);
-        } catch (Exception e) {
-            if (LOGGER.isLoggable(Level.FINE)) {
-                LOGGER.log(Level.FINE, e.getLocalizedMessage());
-            }
+        } catch (Throwable e) {
+            LOGGER.log(Level.SEVERE, "Error setting up GranuleDescriptor", e);
+            throw new RuntimeException("Error setting up GranuleDescriptor.", e);
         }
```
Comment (aaime) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java:159]: From ignored exception to throwing? That's not backwards compatible... also, what is the relationship with the issue at hand?

**Review by aaime — COMMENTED** [review]
From the ticket it seems that there were classpath setup issues or incompatible libraries loaded in the classpath, causing the issue.

PAM files are sidecars that can be used with all sorts of input files, they are not images, but statistical summaries or the raster contents. 

The comment "... PAMParser has no meaning. Since the mosaic doesn’t consist of that kind of files." seems to have been written in a moment of distress, with no understanding of what a PAM file is (no mosaic can "consist of that kind of files" indeed).

An error such as " java.lang.ClassNotFoundException: javax.xml.bind.JAXBException" 
seems to simply point at a missing dependency.

It seems to me what you want is to have the JAXB dependency be made optional.

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java (hunk 1 above)
Comment (roarbra) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java:159]: I'm willing to skip the throwing, but I would keep the logging.

**Comment by roarbra** [comment]
> The comment "... PAMParser has no meaning. Since the mosaic doesn’t consist of that kind of files." seems to have been written in a moment of distress, with no understanding of what a PAM file is (no mosaic can "consist of that kind of files" indeed).

That sentence could be interpreted in all kind of ways, and doesn't necessarily mean that I say it out of distress. Just that I have looked up what a PAM file is and see that the image files of my mosaic doesn't have such files.

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java [hunk 2]
```diff
@@ -641,7 +639,16 @@ private void checkPamDataset() throws IOException {
...
+        if (auxFile.exists()) {
+            if (pamParser == null) {
+                try {
+                    pamParser = PAMParser.getInstance();
+                } catch (Throwable e) {
+                    throw new RuntimeException("Couldn't initialize PAM parser.", e);
+                }
+            }
+            pamDataset = pamParser.parsePAM(auxFile);
+        }
```
[the remark this answers is not in the document]
Comment (dromagnoli) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java:651]: I agree. Nothing special to report here.

**Pull request merged**